File: recursion/cppCode/sumOfNumber.cpp
#include <iostream>

using namespace std;

int sum(int n) {
    if(n < 1) return 0;
    return n + sum(n - 1);
}

void printSum(int i, int sum) {
    if(i < 1) {
        cout << sum << endl;
        return;
    }
    printSum(i - 1, sum + i);
}

int factorial(int n) {
    if(n == 0) return 1;
    return n * factorial(n - 1);
}

int main() {
int res = sum(3);
cout << res << endl;

printSum(5, 0);

int fact = factorial(4);
cout << fact << endl;
}

File: recursion/pythonCode/sumOfNumber.py
def sumOfNum(n):
    if(n < 1): return 0
    return n + sumOfNum(n - 1)

def parameterizedSum(i, sum):
    if i < 1 :
        print(sum)
        return
    parameterizedSum(i - 1, sum + i)

def factorial(n):
    if n == 0: return 1
    return n * factorial(n - 1)

if __name__ == "__main__":
    print(sumOfNum(3))
    parameterizedSum(5, 0)
    print(factorial(5))